templates/php.yml

```diff
@@ -1,26 +1,80 @@
 # PHP
-# Test and package your PHP project.
-# Add steps that run tests, save build artifacts, deploy, and more:
+# Build, package and deploy your PHP project.
+# Add steps that run tests and more:
 # https://docs.microsoft.com/azure/devops/pipelines/languages/php
 
 trigger:
 - master
 
-pool:
-  vmImage: 'ubuntu-latest'
-
 variables:
-  phpVersion: 7.2
-
-steps:
-- script: |
-    sudo update-alternatives --set php /usr/bin/php$(phpVersion)
-    sudo update-alternatives --set phar /usr/bin/phar$(phpVersion)
-    sudo update-alternatives --set phpdbg /usr/bin/phpdbg$(phpVersion)
-    sudo update-alternatives --set php-cgi /usr/bin/php-cgi$(phpVersion)
-    sudo update-alternatives --set phar.phar /usr/bin/phar.phar$(phpVersion)
-    php -version
-  displayName: 'Use PHP version $(phpVersion)'
-
-- script: composer install --no-interaction --prefer-dist
-  displayName: 'composer install'
+  # Azure Resource Manager connection created during pipeline creation
+  azureSubscription: '{{ azureRmConnection.Id }}'
+  
+  # Web app name
+  webAppName: '{{ webAppName }}'
+
+  # Agent VM image name
+  vmImageName: 'ubuntu-latest'
+
+  # Environment name
+  environmentName: "$(Build.DefinitionName)-Prod"
+
+  # Root folder under which your composer.json file is available.
+  rootFolder: $(System.DefaultWorkingDirectory)
+
+stages:
+- stage: Build
+  displayName: Build stage
+  variables:
+    phpVersion: 7.2
+  jobs:
+  - job:
+    pool:
+      vmImage: $(vmImageName)
+    steps:
+    - script: |
+        sudo update-alternatives --set php /usr/bin/php$(phpVersion)
+        sudo update-alternatives --set phar /usr/bin/phar$(phpVersion)
+        sudo update-alternatives --set phpdbg /usr/bin/phpdbg$(phpVersion)
+        sudo update-alternatives --set php-cgi /usr/bin/php-cgi$(phpVersion)
+        sudo update-alternatives --set phar.phar /usr/bin/phar.phar$(phpVersion)
+        php -version
+      workingDirectory: $(rootFolder)
+      displayName: 'Use PHP version $(phpVersion)'
+    
+    - script: composer install --no-interaction --prefer-dist
+      workingDirectory: $(rootFolder)
+      displayName: 'composer install'
```

Comment:
I assume, composer install wont thow any error, if `composer.json` or `composer.phar` not found.

Reply:
It does throw an error if composer.json is not present. That's a good point.

I'm not sure if we have to add a if else check though. As, most of the repos would have composer.json. In some flows, we might not find composer.json when your application code is probably under a sub folder. If the task fails, user would know that this needs to be corrected. In that case, he would edit the root folder value. If we add a if else check, we would not be surfacing this very well. Let me check with PM on this one.

Reply:
We have decided to let the composer install fail. User can remove the task if there is no composer.json.